src/scmrepo/git/__init__.py

```diff
@@ -327,12 +327,18 @@ def fetch_refspecs(
         progress: Optional[Callable[["GitProgressEvent"], None]] = None,
         **kwargs,
     ) -> typing.Mapping[str, SyncStatus]:
+        from urllib.parse import urlparse
+
         from .credentials import get_matching_helper_commands
 
-        if "dulwich" in kwargs.get("backends", self.backends.backends) and any(
-            get_matching_helper_commands(url, self.dulwich.repo.get_config_stack())
-        ):
-            kwargs["backends"] = ["dulwich"]
+        if "dulwich" in kwargs.get("backends", self.backends.backends):
+            credentials_helper = any(
+                get_matching_helper_commands(url, self.dulwich.repo.get_config_stack())
+            )
+            parsed = urlparse(url)
+            ssh = parsed.scheme in ("git", "git+ssh", "ssh") or url.startswith("git@")
```

Comment:
This should maybe be handled here: https://github.com/iterative/scmrepo/blob/e8de01e07edeacae33a9552a1005152b5a153b2a/src/scmrepo/git/backend/pygit2.py#L463-L464

But this may be a different problem entirely if pygit2+SSH is not working on linux/mac. libgit2/pygit2 are supposed to support SSH everywhere except windows.

Reply:
I guess the issue is that while it supports SSH, libgit2 doesn't parse openssh configs

https://libgit2.org/docs/guides/authentication/

So we probably need to be using `pygit2.RemoteCallbacks` and loading the keyfile or user/pass ourselves (we can use asyncssh for this)